features/steps/publisher.go

```diff
@@ -2,32 +2,58 @@ package steps
 
 import (
 	"context"
-	"fmt"
+	"github.com/chromedp/cdproto/network"
 	"github.com/chromedp/chromedp"
+	"net/http"
 )
 
 type Publisher struct {
 	fakeApi   *FakeApi
 	chromeCtx context.Context
+	cookies   []*network.Cookie
 }
 
 func NewPublisher(api *FakeApi, ctx context.Context) *Publisher {
 	return &Publisher{fakeApi: api, chromeCtx: ctx}
 }
 
-func (p *Publisher) logIn(username string) error {
+func (p *Publisher) signIn(username string) error {
+	var cookies []*http.Cookie
 
-	// Here we are setting the fake data that would be returned from the identity-api
-	// with the permissions requires to create a new collection
-	p.fakeApi.fakeHttp.NewHandler().Post("/login").Reply(200).Body([]byte(`faketoken`))
-	p.fakeApi.setJsonResponseForGet("/permission", fmt.Sprintf(`{"email":"%s","admin":true,"editor":true}`, username))
+	if username == "not.a.publisher@ons.gov.uk" {
+		p.fakeApi.setJsonResponseForPost("/tokens", "", 401)
+	} else {
+		p.fakeApi.setJsonResponseForPost("/tokens", "{\"expirationTime\": \"2020-01-01 00-00-01Z\"}", 200)
+		cookies = append(cookies, GenerateCookie("user_token", "fakeAuthorizationToken", ""))
+		cookies = append(cookies, GenerateCookie("id_token", "fakeIDToken", ""))
+		cookies = append(cookies, GenerateCookie("refresh_token","fakeRefreshToken", ""))
+	}
 
 	err := chromedp.Run(p.chromeCtx,
+		SetCookies(cookies),
 		chromedp.Navigate("http://localhost:8080/florence/login"),
 		chromedp.WaitVisible(`#app`),
 		chromedp.SendKeys("#email", username),
 		chromedp.SendKeys("#password", "anything"),
 		chromedp.Click(".form button"),
+		p.assignUserCookies(),
+	)
+
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (p *Publisher) signOut() error {
```

Comment:
So sign out is actually currently broken.
We did signOut before sign in stories, since doing the signIn stories we decided to set cookies server-side for better security. However, since moving it to the backend it means that the signout functionality won't delete cookies as it attempts to do this with JS. So this feature test should be failing at the moment, until that is fixed.

Reply:
Although a little confused about how this is passing at the moment as we are setting the cookies in the test via `Given I have Auth tokens` then we are deleting the cookies in the codebase by `I sign out` but this would be being done with JS right, yet I can see the cookies in the test say `httpOnly`.